spinalcordtoolbox/straightening.py

```diff
@@ -478,10 +477,7 @@ def straighten(self):
                 coord_in_planes_curved = centerline.get_in_plans_coordinates(projected_points_curved,
                                                                              nearest_indexes_curved)
 
-                coord_curved2straight = centerline_straight.points[lookup]
-                coord_curved2straight[:, 0:2] += coord_in_planes_curved[:, 0:2]
-                coord_curved2straight[:, 2] += distances_curved
-
+                coord_curved2straight = centerline_straight.get_inverse_plans_coordinates(coord_in_planes_curved, lookup)
```

Comment:
Alright, since the Z differences are large, I decided to take a visual look at the result of this code change. It looks like this was not just a refactor, but actually a crucial part of fixing the shearing problem!

Reply:
!!! Wonderful!